Update userfaultfd to match the version packaged in Fedora, which fixes
unrecognized ioctl failures.

--- a/src/vmm/Cargo.toml
+++ b/src/vmm/Cargo.toml
@@ -25,7 +25,7 @@
 timerfd = "1.5.0"
 thiserror = "1.0.32"
 displaydoc = "0.2.4"
-userfaultfd = "0.7.0"
+userfaultfd = "0.8.1"
 vhost = { version = "0.10.0", features = ["vhost-user-frontend"] }
 vm-allocator = "0.1.0"
 vm-superio = "0.7.0"
